Make single ticket embeds smaller

violine1101 opened the pull request:
This PR changes some things about single ticket embeds, so that they are not as large as before.

* Screenshots are now always thumbnails
* Text of the ticket is displayed as short as possible

NeunEinser commented:
If we want to actually translate code blocks, that would be possible too, but it really only makes sense if the restriction of `\n` is somehow tweaked.
```ts
description = description.replace(
	/{code(?::([A-Za-z#+]+)(?:\|.+?)*|.+?)?}((?:.|\n)+?){code}/gm,
	(_, lang: String, code: String) => 
		`\`\`\`${ lang?.toLowerCase().replace('c#', 'cs').replace('c++', 'cpp') ?? 'java' }\n${ code.trim() }\n\`\`\``
	
)
```
Without line restriction:
![image](https://user-images.githubusercontent.com/12124394/89593828-73ccb400-d850-11ea-9cba-308074861bb1.png)

violine1101 commented on src/mentions/SingleMention.ts:
```diff
@@ -70,8 +70,36 @@ export class SingleMention extends Mention {
+		// unify line breaks
 		description = description.replace( /^\s*[\r\n]/gm, '\n' );
+
+		// remove headings
+		description = description.replace( /^\s*h\d.*$/mi, '' );
+		description = description.replace( /^\*.+\*$/gm, '' );
```
These lines are usually headings, which is why I'm removing them. Though I guess there are cases where it's still important for them to stay.

violine1101 commented on src/mentions/SingleMention.ts:
```diff
@@ -70,8 +70,36 @@ export class SingleMention extends Mention {
+		// remove all sections except for the first
+		description = description.replace( /\nh\d[\s\S]*$/i, '' );
+
+		// replace code block format
```
Whoops, yes, you're right

violine1101 commented:
After I made this PR, I came across this NPM package https://www.npmjs.com/package/jira2md that automatically converts JIRA markdown to generic markdown which should work in Discord too.

I think it's probably a better idea to rely on that package instead of reimplementing all of it here.

NeunEinser commented:
> 
> 
> After I made this PR, I came across this NPM package https://www.npmjs.com/package/jira2md that automatically converts JIRA markdown to generic markdown which should work in Discord too.
> 
> I think it's probably a better idea to rely on that package instead of reimplementing all of it here.

Well, not entirely. Discord doesn't support a bunch of things general markdown does support, such as tables and headings. But it does seem like a very good starting point.

violine1101 commented:
I've extracted everything we need from that package into a separate util function and modified it to Discord markdown (and added multiline quotes). I think this is probably the way to go.
